### PR TITLE
how-to: Node-RED filter node

### PR DESCRIPTION
**Pull request by ZJvandeWeg**
## Description

Had a few minutes in my day, so wrote this up. Think it doesn't warrant any social media push, but might be great to be indexed for the community to Google how to use the Filter node.

If it needs moving later, we can use a redirect.

### CODE REVIEW

File: src/blog/2023/06/node-explained-filter.md [hunk 1]
```diff
@@ -0,0 +1,108 @@
+---
+title: "Nodes explained: Filter"
+subtitle: The filter node is incredibly versatile, let's go over some use-cases
+description: Forwarding or blocking messages based on values, change of them, or lack of change.
+date: 2023-06-15
+authors: ["zeger-jan-van-de-weg"]
+image:
```
Comment (ZJvandeWeg) [inline src/blog/2023/06/node-explained-filter.md:7]: Not sure if this is worth the time for an art request? Or if a generated image is fine?

Comment (iskerrett) [reply src/blog/2023/06/node-explained-filter.md:7]: I agree it is not worth it.

File: src/blog/2023/06/node-explained-filter.md [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+tags:
+  - posts
+  - how-to
+---
+
+The Filter node, previously called "Report by Exception", has two modes of
+operation, called deadband and narrowband.
+It is commonly used in applications where sensors constantly report data and only
+significant changes need to be communicated to the control system. Or in other
+cases the lack of change needs reporting.
```
Comment (iskerrett) [inline src/blog/2023/06/node-explained-filter.md:17]: The last sentence is a bit awkward.  I would actually delete it or maybe reword something like

'In other cases, alerts are required if sensor data doesn't change over a period of time.'

Comment (iskerrett) [reply src/blog/2023/06/node-explained-filter.md:17]: Reading below it doesn't seem you have an example that illustrates the use case in the last sentence?

**Review by iskerrett — COMMENTED** [review]
Really good article!

**Comment by Hisma** [comment]
It sounds useful and this is very common use-case in industrial automation.  IE, when polling analog values, we don't want to send updates every time the value changes, as it could be bouncing around. Having a hysteresis or adjustable deadband to control when a data is historized and not spamming your historian with a bouncing analog value.    I think expanding on how this feature can be useful for lightening the load on the historian/database while still capturing meaningful data would add more clarity why you want this capability.

File: src/blog/2023/06/node-explained-filter.md [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+authors: ["zeger-jan-van-de-weg"]
+image:
+tags:
+  - posts
+  - how-to
+---
+
+The Filter node, previously called "Report by Exception", has two modes of
+operation, called deadband and narrowband. These modes allow users to limit network
+traffic or write operations to historians. Or limit reporting of values outside
```
Comment (iskerrett) [inline src/blog/2023/06/node-explained-filter.md:15]:
```suggestion
traffic, write operations to historians, or limit reporting of values outside
```

File: src/blog/2023/06/node-explained-filter.md [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+a range that is worth reporting on. It's very versitile, once you understand the
```
Comment (iskerrett) [inline src/blog/2023/06/node-explained-filter.md:16]:
```suggestion
a range that is worth reporting on. It's very versatile, once you fully understand the
```

File: src/blog/2023/06/node-explained-filter.md [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+node fully.
```
Comment (iskerrett) [inline src/blog/2023/06/node-explained-filter.md:17]:
```suggestion
node.
```

File: src/blog/2023/06/node-explained-filter.md [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+
+<!--more-->
+
+The Filter node is part of the core nodes in Node-RED, meaning they're installed
```
Comment (iskerrett) [inline src/blog/2023/06/node-explained-filter.md:21]:
```suggestion
The Filter node is part of the core nodes in Node-RED, meaning it is installed
```

**Review by iskerrett — APPROVED** [review]

File: src/blog/2023/06/node-explained-filter.md [hunk 7]
```diff
@@ -0,0 +1,108 @@
+---
+title: "Nodes explained: Filter"
+subtitle: The filter node is incredibly versatile, let's go over some use-cases
+description: Forwarding or blocking messages based on values, change of them, or lack of change.
+date: 2023-06-15
```
Comment (ZJvandeWeg) [inline src/blog/2023/06/node-explained-filter.md:5]:
```suggestion
date: 2023-06-16
```
